core/src/test/java/org/apache/calcite/rex/RexProgramTest.java

```diff
@@ -3109,4 +3110,59 @@ private SqlSpecialOperatorWithPolicy(String name, SqlKind kind, int prec, boolea
   @Test void testSimplifyVarbinary() {
     checkSimplifyUnchanged(cast(cast(vInt(), tVarchar(true, 100)), tVarbinary(true)));
   }
+
+  @Test public void testSimplifyBoolean() {
+    final RelDataType intType = typeFactory.createSqlType(SqlTypeName.INTEGER);
+    final RelDataType rowType = typeFactory.builder()
+        .add("a", intType)
+        .add("b", intType)
+        .add("c", intType)
+        .add("d", intType)
+        .build();
+    final RexDynamicParam range = rexBuilder.makeDynamicParam(rowType, 0);
+    final RexNode aRef = rexBuilder.makeFieldAccess(range, 0);
+    final RexNode bRef = rexBuilder.makeFieldAccess(range, 1);
+    final RexNode cRef = rexBuilder.makeFieldAccess(range, 2);
+    final RexNode dRef = rexBuilder.makeFieldAccess(range, 3);
```

Comment:
Please use `vBool(...)`, `vInt(...)` for testing variables. Then you do not need to declare them as variables.

`aRef` is bad naming since it provides no clue on the type of the variable (e.g. int vs bool).

Reply:
Ok, latter I'will take a look.

Reply:
There are full of aRef in this test, do we need refactor them all in a new JIRA?

Reply:
`aRef` is the old way of writing tests.
You'd better check the code creation time before copy-pasting it.

>There are full of aRef in this test, do we need refactor them all in a new JIRA?

A complete rewrite might result in merge conflicts with doubtful gains, so I would refrain from mass-rewriting RexProgramTest while search/sarg feature is in a half-working state.

Reply:
sure

Reply:
resolved